robolectric/src/test/java/org/robolectric/shadows/ShadowWallpaperManagerTest.java

```diff
@@ -576,4 +581,35 @@ private static byte[] getBytesFromBitmap(Bitmap bitmap) {
     bitmap.compress(Bitmap.CompressFormat.PNG, /* quality= */ 0, stream);
     return stream.toByteArray();
   }
+
+  @Test
+  @Config(minSdk = O)
+  public void wallpaperManager_activityContextEnabled_retrievesSameWallpaper() {
+    String originalProperty = System.getProperty("robolectric.createActivityContexts", "");
+    System.setProperty("robolectric.createActivityContexts", "true");
+    Activity activity = null;
+    try {
+      WallpaperManager applicationWallpaperManager =
+          (WallpaperManager)
+              ApplicationProvider.getApplicationContext()
+                  .getSystemService(Context.WALLPAPER_SERVICE);
+
+      activity = Robolectric.setupActivity(Activity.class);
+      WallpaperManager activityWallpaperManager =
+          (WallpaperManager) activity.getSystemService(Context.WALLPAPER_SERVICE);
+
+      assertThat(applicationWallpaperManager).isNotSameInstanceAs(activityWallpaperManager);
```

Comment:
> Ideally, Robolectric should keep the same behavior for instance comparing as Android Emulator. By checking above ContextTest, they should be the same from Android S. And looks like there are some extra change should be made for Robolectric.

Will it be better to compare their WallpaperInfo objects to ensure they represent the same wallpaper.
Instead of directly comparing the WallpaperManager instances, which might differ due to Robolectric's internal behavior and remove this assertion " assertThat(applicationWallpaperManager).isNotSameInstanceAs(activityWallpaperManager);"